admin-dev/themes/new-theme/js/components/form/submittable-input.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+const {$} = window;
+
+/**
+ * Activates, deactivates, shows, hides submit button inside an input
+ * (depending if input was changed comparing to initial value)
+ * After button is clicked, component fires the callback function which was provided to constructor.
+ */
+export default class SubmittableInput {
+  /**
+   * @param {String} wrapperSelector
+   * @param {Function} callback
+   *
+   * @returns {{}}
+   */
+  constructor(wrapperSelector, callback) {
+    this.inputSelector = '.submittable-input';
+    this.callback = callback;
+    this.wrapperSelector = wrapperSelector;
+    this.buttonSelector = '.check-button';
+
+    this.init();
+
+    return {};
+  }
+
+  /**
+   * @private
+   */
+  init() {
+    const inputs = `${this.wrapperSelector} ${this.inputSelector}`;
+
+    $(document).on('focus', inputs, (e) => {
+      this.refreshButtonState(e.currentTarget, true);
+    });
+    $(document).on('input mouseleave', inputs, (e) => {
+      this.refreshButtonState(e.currentTarget);
+    });
+    $(document).on('click', `${this.wrapperSelector} ${this.buttonSelector}`, (e) => {
+      this.submitInput(e);
+    });
+    $(document).on('keyup', this.inputSelector, (e) => {
+      if (e.keyCode === 13) {
+        e.preventDefault();
+        this.submitInput(e);
+      }
+    });
+  }
+
+  /**
+   * @private
+   */
+  submitInput(e) {
+    const input = this.findInput(e.currentTarget);
+    this.callback(input);
```

Comment:
This works fine, except we don't handle errors at all So I thought instead of using a callback function we should use a Promise instead, this way you can handle both cases:

```js
this.callbackPromise
    .then(() => {
        $(input).data('initial-value', input.value);
        this.toggleButtonVisibility(e.currentTarget, false);
    })
    .catch(() => {
        // display growl error
    })
```

So the callback function is like a "Promise factory" (hence the name callbackPromise) which is implied by the `async` keyword I don't know if `async function` is a valid type

@NeOMakinG what do you think about this idea?

Reply:
And since the service `updateListedCombination` already returns a Promise it shouldn't be too complex to adapt You just need to use the Promise itself in the constructor, instead of using `await` in a callback function

Reply:
This way we can add an "error state" when the callback fails, not in this PR but in a next one along with the loading state